app/src/androidTest/java/com/appunite/loudius/LoginScreenTest.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2023 AppUnite S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.appunite.loudius
+
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.appunite.loudius.ui.login.LoginScreen
+import com.appunite.loudius.ui.theme.LoudiusTheme
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@HiltAndroidTest
+class LoginScreenTest {
+
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val composeTestRule = createAndroidComposeRule<TestActivity>()
```

Comment:
chore: if you'll use `val composeTestRule = createComposeRule()` instead of `createAndroidComposeRule<TestActivity>()` then you'll not need `TestActivity`, so you can skip changes in the AndroidManifest.xml file.

Reply:
This is true as long as this screen does not have a view model. This version of the code does not include your changes to check if a device is xiaomi, but I checked it on a test branch where I merged it with your code and it wasn't working.

( https://stackoverflow.com/questions/71344878/real-viewmodel-with-composable-injected-by-hilt-in-instrumentation-test-zero-arg?noredirect=1&lq=1 )